src/hyperbolic_system.hpp

```diff
@@ -34,13 +34,27 @@ namespace quokka
 enum redoFlag { none = 0, redo = 1 };
 } // namespace quokka
 
+// Define enum for slope limiter type
+enum SlopeLimiter { minmod = 0, MC };
+
 using array_t = amrex::Array4<amrex::Real> const;
 using arrayconst_t = amrex::Array4<const amrex::Real> const;
 
 /// Class for a hyperbolic system of conservation laws
 template <typename problem_t> class HyperbolicSystem
 {
       public:
+	template <SlopeLimiter limiter> static auto SlopeFunc(amrex::Real x, amrex::Real y) -> amrex::Real
+	{
+		if (limiter == SlopeLimiter::minmod) {
+			return minmod(x, y);
+		}
+		if (limiter == SlopeLimiter::MC) {
+			return MC(x, y);
+		}
+		throw std::invalid_argument("Invalid slope limiter specified.");
```

Comment:
We should **never** throw exceptions. They do not work on GPU. Since this is a template argument, you can use `static_assert` here.

Reply:
Done.